### PR TITLE
Chapter1

### PR DESCRIPTION
**Pull request by minjipack**
### Solutions for Chapter {NUMBER}

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -449,7 +454,10 @@ Implement the function that takes an integer value and returns the next 'Int'.
   function body with the proper implementation.
 -}
 next :: Int -> Int
-next x = error "next: not implemented!"
+next x = x + 1
+{-
+greater than max -> minimum
```
Comment (vrom911) [inline src/Chapter1.hs:459]: Good catch!
We will discuss this in future chapters 😄

File: src/Chapter1.hs [hunk 2]
```diff
@@ -632,7 +644,24 @@ Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
 
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int  
+sumLast2 n = div lastTwoDigits 10 + mod lastTwoDigits 10
```
Comment (vrom911) [inline src/Chapter1.hs:648]: That is a wonderful solution! 👏🏼  You correctly noticed that it is the `div` and `mod`, cool 😎 

One hint to make your solution even shorter: you can see that you use both:

```haskell
mod x 10
div x 10
```
The standard library has the [`divMod`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Prelude.html#v:divMod) function, that actually combines inside both `div` and `mod`. And this is exactly what you use!.

So you could write it this way:

```haskell
(x, y) = divMod ... 10
```

You can see how we could pattern match on the pair 🙂

File: src/Chapter1.hs [hunk 3]
```diff
@@ -653,8 +682,10 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
-
+firstDigit :: Int -> Int
+firstDigit n = if absN < 10 then mod absN 10 else firstDigit (div absN 10)
```
Comment (vrom911) [inline src/Chapter1.hs:686]: If `abs n < 10` you don't need to do `mod` on that 🙂 
So you can simplify it a bit:
```suggestion
firstDigit n = if absN < 10 then absN else firstDigit (div absN 10)
```

**Review by vrom911 — APPROVED** [review]
Great work! Very nice solutions! 🎉 

And awesome work on the advanced task as well 👏🏼

File: src/Chapter1.hs (hunk 3 above)
Comment (minjipack) [reply src/Chapter1.hs:686]: Oh yes! You are absolutely right. Thank you for your comment :)

File: src/Chapter1.hs (hunk 1 above)
Comment (minjipack) [reply src/Chapter1.hs:459]: Looking forward to future chapters  🍡

File: src/Chapter1.hs (hunk 2 above)
Comment (minjipack) [reply src/Chapter1.hs:648]: Thanks! `divMod` is a cool thing to know! I don't think I can come up with a shorter answer with `divMod` ...  :( The following is what I came up with.

```
sumLast2 :: Int -> Int
sumLast2 n =
 let (secondLast, last) = divMod n 10
 in (mod secondLast 10) + last
```

**Review by vrom911 — APPROVED** [review]
These changes are awesome! Ready to go! 🚢

**Comment by vrom911** [comment]
I am going to close this PR as we can't merge it into the repo, but we are hoping to see you in the next Chapter 🎉

**Pull request closed without merging**